drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -61,9 +64,14 @@ class Hmis::Hud::Client < Hmis::Hud::Base
   end
 
   scope :with_access, ->(user, *permissions, **kwargs) do
-    return none unless user.permissions?(*permissions, **kwargs)
+    pids = Hmis::Hud::Project.with_access(user, *permissions, **kwargs).pluck(:id)
+    base_scope = left_outer_joins(:projects, :wip, :data_source)
+
+    unenrolled_scope = user.permissions?(*permissions, **kwargs) ? unenrolled.left_outer_joins(:data_source).merge(GrdaWarehouse::DataSource.hmis(user)) : none
+    enrolled_scope = base_scope.where(p_t[:id].in(pids))
+    wip_scope = base_scope.where(wip_t[:project_id].in(pids))
 
-    joins(:data_source).merge(GrdaWarehouse::DataSource.hmis(user))
+    unenrolled_scope.or(enrolled_scope).or(wip_scope)
```

Comment:
This is not returning a unique list of clients, which I think it should. Do yall agree? Whats the most efficient way to to that? Elsewhere we pluck IDs and do another lookup.

Reply:
That was going to be my suggestion.

Reply:
or `where(id: unenrolled_ids + enrolled_ids + wip_ids)` which is easier to read for me.
`or` clauses can be slow.  If the plucking of the ids is significantly faster that seems like a good approach to me.

Reply:
Oh, I was confused, I was responding only to the performance question below, but yeah, maybe it'll solve both.

Reply:
great